go/ai/document.go

```diff
@@ -31,11 +31,11 @@ type Document struct {
 // A Part is one part of a [Document]. This may be plain text or it
 // may be a URL (possibly a "data:" URL with embedded data).
 type Part struct {
-	kind         partKind
-	contentType  string        // valid for kind==blob
-	text         string        // valid for kind∈{text,blob}
-	toolRequest  *ToolRequest  // valid for kind==partToolRequest
-	toolResponse *ToolResponse // valid for kind==partToolResponse
+	Kind         partKind      `json:"kind,omitempty"`
```

Comment:
Ah yeah, missed this. Unfortunately, had to undo all of this nice visibility control here because there's no way to generate the schema correctly any other way from what I can tell.